### PR TITLE
add startup and readiness probes to sts component specs

### PR DESCRIPTION
**Pull request by kruftik**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

**Comment by koct9i** [comment]
For the reference:
https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes

File: pkg/components/server.go [hunk 1]
```diff
@@ -218,6 +228,27 @@ func (s *serverImpl) rebuildStatefulSet() *appsv1.StatefulSet {
 				Command:      []string{s.binaryPath, "--config", path.Join(consts.ConfigMountPoint, fileNames[0])},
 				VolumeMounts: volumeMounts,
 				Resources:    s.instanceSpec.Resources,
+				StartupProbe: &corev1.Probe{
```
Comment (koct9i) [inline pkg/components/server.go:231]: Not sure why we need startup probe here.

File: pkg/components/server.go [hunk 2]
```diff
@@ -17,6 +18,15 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+const (
+	initialProbesDelaySeconds = 90
```
Comment (koct9i) [inline pkg/components/server.go:22]: Initial delay isn't required either.

File: pkg/components/server.go [hunk 3]
```diff
@@ -17,6 +18,15 @@ import (
...
+
+	probesTimeoutSeconds = 3
+	probesPeriodSeconds  = 15
+
+	readinessProbeHTTPPath = "/solomon/status"
```
Comment (l0kix2) [inline pkg/components/server.go:27]: solomon path shouldn't be hardcoded in the public code, it should be configured in spec or somehow else.

File: pkg/components/server.go [hunk 4]
```diff
@@ -17,6 +18,15 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+const (
+	initialProbesDelaySeconds = 90
+
+	probesTimeoutSeconds = 3
+	probesPeriodSeconds  = 15
```
Comment (l0kix2) [inline pkg/components/server.go:25]: I suppose these params also should be in spec and not hardcoded.

**Review by l0kix2 — CHANGES_REQUESTED** [review]
I suggest creating some `ProbeSpec` struct in types and  fields of that type in the `InstanceSpec` struct which is reused for most of the components.
Not sure if fields should be named `StartupProbe` & `ReadinessProbe` though, better not clash with k8s names, maybe `StartupProbeTemplate` or smth would do?

File: pkg/components/server.go (hunk 3 above)
Comment (zlobober) [reply pkg/components/server.go:27]: Is there any specific need to use solomon?
I think it is ok to use host:monitoring_port/orchid/service, it is just fine. And I do not see any issue to hardcode this particular address in the code, it is a detail of YT's implementation, which is shared by absolutely all C++ server components.

**Comment by kruftik** [comment]
@koct9i @l0kix2 @zlobober pls take another look at the changes

**Review by l0kix2 — APPROVED** [review]
LGTM 
@koct9i @zlobober do you have more comments?

**Comment by koct9i** [comment]
Ok, LGTM.

**Review by koct9i — APPROVED** [review]

**Pull request merged**